packages/phpstan-rules/src/Rules/PreventDoubleSetParameterRule.php

```diff
@@ -23,68 +21,79 @@ final class PreventDoubleSetParameterRule extends AbstractSymplifyRule
     /**
      * @var string
      */
-    public const ERROR_MESSAGE = 'Set param value is duplicated, use unique value instead';
+    public const ERROR_MESSAGE = 'Set param "%s" value is duplicated, use unique value instead';
 
     /**
-     * @var NodeFinder
+     * @var SimpleNameResolver
      */
-    private $nodeFinder;
+    private $simpleNameResolver;
 
-    public function __construct(NodeFinder $nodeFinder)
-    {
-        $this->nodeFinder = $nodeFinder;
+    /**
+     * @var SymfonyPhpConfigClosureAnalyzer
+     */
+    private $symfonyPhpConfigClosureAnalyzer;
+
+    /**
+     * @var array<string, string[]>
+     */
+    private $setParametersNamesByFile = [];
+
+    /**
+     * @var ConstExprEvaluator
+     */
+    private $constExprEvaluator;
+
+    public function __construct(
+        SimpleNameResolver $simpleNameResolver,
+        SymfonyPhpConfigClosureAnalyzer $symfonyPhpConfigClosureAnalyzer,
+        ConstExprEvaluator $constExprEvaluator
+    ) {
+        $this->simpleNameResolver = $simpleNameResolver;
+        $this->symfonyPhpConfigClosureAnalyzer = $symfonyPhpConfigClosureAnalyzer;
```

Comment:
@samsonasik Hi, the rule for set() checks has better design `SymfonyPhpConfigClosureAnalyzer`.
This could be useful for next Symfony config rules.